Lib/test/test_zoneinfo/test_zoneinfo.py

```diff
@@ -1717,11 +1718,31 @@ def test_env_variable_relative_paths(self):
         for input_paths, expected_paths in test_cases:
             path_var = os.pathsep.join(input_paths)
             with self.python_tzpath_context(path_var):
+                with self.subTest("module-level warning", path_var=path_var):
+                    with CleanImport("zoneinfo", "zoneinfo._tzpath"):
+                        with warnings.catch_warnings(record=True) as wlog:
+                            import zoneinfo
+                    self.assertEqual(len(wlog), 1)
+                    # Since we use import hacks, we cannot just use `isinstance`
+                    self.assertEqual(
+                        type(wlog[0].message).__qualname__,
+                        "InvalidTZPathWarning",
+                    )
+                    # It should represent the current file:
+                    self.assertTrue(
+                        wlog[0].filename.endswith("test_zoneinfo.py"),
+                        msg=wlog[0].filename,
+                    )
+
                 with self.subTest("warning", path_var=path_var):
                     # Note: Per PEP 615 the warning is implementation-defined
                     # behavior, other implementations need not warn.
-                    with self.assertWarns(self.module.InvalidTZPathWarning):
+                    with self.assertWarns(self.module.InvalidTZPathWarning) as w:
                         self.module.reset_tzpath()
+                    self.assertTrue(
+                        w.warnings[0].filename.endswith("test_zoneinfo.py"),
+                        msg=w.warnings[0].filename,
+                    )
```

Comment:
Why not simply:
```suggestion
                    self.assertEqual(w.warnings[0].filename, __file__)
```
?

Reply:
I've already done that in https://github.com/python/cpython/pull/106234/commits/ccdbaa61eeb64d3eb9f5584f09f3524a1d1c492e :)